### ISSUE TITLE
LINKMAP doesn't work with subquery

### ISSUE THREAD

**Issue by freeart** [orientechnologies/orientdb#8646]
## Expected behavior  
Should work like 
```java
INSERT INTO Test (mymap) VALUES ({'A-1': #xx:x});
```
```LINKSET, LINKLIST, LINK, EMBEDDEDSET, EMBEDDEDLIST, EMBEDDED``` work fine with subquery syntax

## Actual behavior  
```com.orientechnologies.orient.core.exception.OValidationException: The field 'Test.mymap' has been declared as LINKMAP but the value is not a record or a record-id DB name="nvd" Error Code="4"```  

## Steps to reproduce  
```java
CREATE CLASS Test;
CREATE CLASS TestMap;
CREATE PROPERTY Test.mymap LINKMAP TestMap;

INSERT INTO TestMap (name) VALUES ('test');
INSERT INTO Test (mymap) VALUES ({'A-1': (SELECT FROM TestMap WHERE name = 'test')});
```

**Comment by luigidellaquila** [comment]
Hi @freeart 

I just pushed a fix to both 3.0.x and develop branches, now it should be OK

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Mmm... it seems I have to revert the fix on develop branch, there is one failing test.
Anyway, I'm working on it, I'll provide a new fix soon
